### PR TITLE
(Place) Removed useless motion of arm and improved docs

### PR DESCRIPTION
**Pull request by P-ict0**
Current implementation: Robot goes to the `prepare_place` pose BEFORE navigating to the place pose.
PR: Robot now doesn't move to the `prepare_place` pose anymore at any moment.

Also changed some asserts to use more correct type checking

### CODE REVIEW

**Review by PetervDooren — APPROVED** [review]
Depending on the implementation used we may need to put prepare place where it is now commented. It would be nice to figure that out fast.

**Comment by PetervDooren** [comment]
It is already broken for robots other than Hero. We don't have the infrastructure to properly support 'any' robot.
What happened here is that the functionality of PUT was implicitly changed due to switching to collision avoidance. As long as the other robots keep a similar functionality it will remain consistent.

**Review by PetervDooren — DISMISSED** [review]

File: robot_smach_states/src/robot_smach_states/manipulation/place.py [hunk 1]
```diff
@@ -167,7 +167,6 @@ def __init__(self, robot, item_to_place, place_pose, arm, place_volume=None, upd
         # Check types or designator resolve types
         assert(item_to_place.resolve_type == Entity or type(item_to_place) == Entity)
         assert(arm.resolve_type == PublicArm or type(arm) == PublicArm)
```
Comment (MatthijsBurgh) [inline robot_smach_states/src/robot_smach_states/manipulation/place.py:169]: Should use `check_resolve_type`

File: robot_smach_states/src/robot_smach_states/manipulation/place.py [hunk 2]
```diff
@@ -165,8 +165,8 @@ def __init__(self, robot, item_to_place, place_pose, arm, place_volume=None, upd
         smach.StateMachine.__init__(self, outcomes=['done', 'failed'])
 
         # Check types or designator resolve types
-        assert(item_to_place.resolve_type == Entity or type(item_to_place) == Entity)
-        assert(arm.resolve_type == PublicArm or type(arm) == PublicArm)
```
Comment (MatthijsBurgh) [inline robot_smach_states/src/robot_smach_states/manipulation/place.py:205]: Sorry, use `check_type`

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]

**Comment by PetervDooren** [comment]
@MatthijsBurgh, the check_type is a correct improvement but it has nothing to do with the changes in this PR. The changes here are fine and can be merged without worrying about changing check type as well. 
If we want to change this check type I suggest doing it in a small PR rather than delaying this one any further.

File: robot_smach_states/src/robot_smach_states/manipulation/place.py [hunk 3]
```diff
@@ -201,9 +165,8 @@ def __init__(self, robot, item_to_place, place_pose, arm, place_volume=None, upd
...
-        #assert(place_volume.resolve_type == str or (type(place_volume) == str))
+        assert(check_type(item_to_place, Entity) or isinstance(item_to_place, Entity))
+        assert(check_type(arm, PublicArm) or isinstance(arm, PublicArm))
```
Comment (MatthijsBurgh) [inline robot_smach_states/src/robot_smach_states/manipulation/place.py:169]:
```suggestion
        check_type(item_to_place, Entity)
        check_type(arm, PublicArm)
```

**Comment by MatthijsBurgh** [comment]
@PetervDooren changes were already made to these lines. So therefore the changes need to be correct and complete.

I applied the changes.

**Comment by PetervDooren** [comment]
> @PetervDooren changes were already made to these lines. So therefore the changes need to be correct and complete.
> 
> I applied the changes.

No they were not. A commented line was removed. That does not make Rodrigo responsible for fixing all the lines near that. That is not how this works.

**Review by PetervDooren — APPROVED** [review]

**Pull request merged**